lib/operations/db_ops.js

```diff
@@ -90,28 +90,27 @@ function addUser(db, username, password, options, callback) {
         // We got an error (f.ex not authorized)
         if (err != null) return handleCallback(callback, err, null);
         // Check if the user exists and update i
-        collection
-          .find({ user: username }, { dbName: options['dbName'] }, finalOptions)
-          .toArray(err => {
-            // We got an error (f.ex not authorized)
-            if (err != null) return handleCallback(callback, err, null);
-            // Add command keys
-            finalOptions.upsert = true;
-
-            // We have a user, let's update the password or upsert if not
-            update(
-              collection,
-              { user: username },
-              { $set: { user: username, pwd: userPassword } },
-              finalOptions,
-              err => {
-                if (count === 0 && err)
-                  return handleCallback(callback, null, [{ user: username, pwd: userPassword }]);
-                if (err) return handleCallback(callback, err, null);
-                handleCallback(callback, null, [{ user: username, pwd: userPassword }]);
-              }
-            );
-          });
+        finalOptions.projection = { dbName: 1 };
```

Comment:
`finalOptions` is subsequently used for the below `updateOne` call, so you have inadvertently added a projection to that command.